apisix/init.lua

```diff
@@ -629,6 +630,12 @@ end
 function _M.http_header_filter_phase()
     core.response.set_header("Server", ver_header)
 
+    local status_code = tonumber(get_var("upstream_status"))
+    if status_code >= 500 and status_code <=599 then
+        core.response.set_header("X-Apisix-Upstream-Status", status_code)
```

Comment:
Would it be better to use `X-APISIX-`? CC @membphis

Reply:
Agree for APISIX

Reply:
Agree +1